SofaKernel/modules/SofaCore/src/sofa/core/objectmodel/DataContentValue.h

```diff
@@ -0,0 +1,150 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <sofa/config.h>
+#include <memory>
+
+namespace sofa::core::objectmodel
+{
+
+/// To handle the Data link:
+/// - CopyOnWrite==false: an independent copy (duplicated memory)
+/// - CopyOnWrite==true: shared memory while the Data is not modified (in that case the memory is duplicated to get an independent copy)
+template <class T, bool CopyOnWrite>
+class DataValue;
+
+template <class T>
+class DataValue<T, false>
```

Comment:
Why DataContentValue.h and not DataValue.h ?

Reply:
Because I didn't wanted to look like the stuff like DataFilename, DataXXX,